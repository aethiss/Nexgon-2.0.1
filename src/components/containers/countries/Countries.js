/* eslint-disable array-callback-return,
    consistent-return,
    react/forbid-prop-types */
import React from 'react'
import PropsTypes from 'prop-types'

// Style
import { ScrollView, TouchableOpacity } from 'react-native'
import { Content, Text, View, Icon, ListItem, Left, Body, Right } from 'native-base'

const Countries = ({ onSelectCountry, countriesList }) => {
  const renderCountryItem = country => (
    <ListItem icon key={`${country.phone}${country.name}`}>
      <Left>
        <Text>{country.emoji}</Text>
      </Left>
      <Body>
        <TouchableOpacity onPress={() => onSelectCountry(country)}>
          <Text>{country.name}</Text>
          <Text note>+{country.phone}</Text>
        </TouchableOpacity>
      </Body>
      <Right>
        <TouchableOpacity onPress={() => onSelectCountry(country)}>
          <Icon name="ios-arrow-forward" style={{ color: 'black' }} />
        </TouchableOpacity>
      </Right>
    </ListItem>
  )

  return (
    <Content>
      <View>
        <Text>Countries</Text>
      </View>
      <View>
        <ScrollView>
          { Object.keys(countriesList).map((country) => {
            if (countriesList[country].phone) {
              return renderCountryItem(countriesList[country])
            }
          })}
        </ScrollView>
      </View>
    </Content>
  )
}

Countries.propTypes = {
  onSelectCountry: PropsTypes.func.isRequired,
  countriesList: PropsTypes.object.isRequired,
}

export default Countries
