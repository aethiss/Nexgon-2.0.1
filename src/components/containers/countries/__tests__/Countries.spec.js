/* eslint-disable no-undef */
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import Countries from '../Countries'

configure({ adapter: new Adapter() })

describe('Register', () => {
  describe('renders', () => {
    it('default render', () => {
      const wrapper = shallow(<Countries onSelectCountry={jest.fn()} />)
      expect(wrapper.dive()).toMatchSnapshot()
    })
  })
})
