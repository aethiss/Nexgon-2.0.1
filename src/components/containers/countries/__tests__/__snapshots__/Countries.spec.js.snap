// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders default render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Content)>
    <Styled(ViewNB)>
      <Styled(Text)>
        Countries
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(Content)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(ViewNB)>
        <Styled(Text)>
          Countries
        </Styled(Text)>
      </Styled(ViewNB)>,
      "style": Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      "virtual": undefined,
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text)>
          Countries
        </Styled(Text)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Countries",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "Countries",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(ViewNB)>
          <Styled(Text)>
            Countries
          </Styled(Text)>
        </Styled(ViewNB)>,
        "style": Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        "virtual": undefined,
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text)>
            Countries
          </Styled(Text)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Countries",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Countries",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
